addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -460,13 +466,17 @@ QUnit.test('new messages separator on posting message', async function (assert)
 
     document.querySelector('.o_ComposerTextInput_textarea').focus();
     await afterNextRender(() => document.execCommand('insertText', false, "hey !"));
-    await afterNextRender(() =>
-        document.querySelector('.o_Composer_buttonSend').click()
-    );
-    assert.containsOnce(
+    await afterNextRender(() => {
+        // need to remove focus from text area to avoid channel_seen
+        document.querySelector('.o_Composer_buttonSend').focus();
```

Comment:
What's the deal with avoiding `channel_seen`? I don't really get it.

Reply:
Initial bug was that you post a message, the new message separator appears and disappears directly.
After investigation it was due to the fact that channel_seen is called as you focus the inputtext and the scroll changes.
So to reproduce the bug without hiding directly the separator you have to send the message and quickly change the focus (it's doable in the interface by using TAB or clicking quickly away after message sent)

So the idea of this test is to ensure that the message separator is not shown at all, for that we need to avoid channel_seen to be called

Reply:
Test should not do this in order to get around a technical issue.
The functional aspect is to not display "new messages" separator when current user post message.
Having focus on composer text input or not should not influence this behaviour at all.

Reply:
If you keep the focus, you don't know if the separator doesn't show because thread has focus or because message is from current partner.

The point is to exclude a possibility to ensure the test is actually covering the fix done.

An alternative way of testing this without messing up with the focus is to post the message with `chat_post` rpc to simulate cross tab sync. This way thread in current browser doesn't have focus in the first place.

Reply:
> If you keep the focus, you don't know if the separator doesn't show because thread has focus or because message is from current partner.
> 
> The point is to exclude a possibility to ensure the test is actually covering the fix done.

Quite technical, such tests should have `[technical]` tag with this explanation.